Sources/Harmony/Data/DataSource/GetNetworkDataSource.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright 2022 Mobile Jazz SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Alamofire
+import Foundation
+
+public class GetNetworkDataSource<T: Decodable>: GetDataSource {
+    
+    private let url: URL
+    private let session: Session
+    private let decoder: JSONDecoder
+    
+    public init(url: URL, session: Session, decoder: JSONDecoder) {
+        self.url = url
+        self.session = session
+        self.decoder = decoder
+    }
+    
+    @discardableResult
+    open func getAll(_ query: Query) -> Future<[T]> {
+        return execute(query)
+    }
+    
+    @discardableResult
+    open func get(_ query: Query) -> Future<T> {
+        return execute(query)
+    }
+    
+    private func execute<K: Decodable>(_ query: Query) -> Future<K> {
+        return Future<K> { resolver in
+          
+            guard let query = validate(query) else {
+                resolver.set(CoreError.QueryNotSupported())
+                return
+            }
+          
+            query
+                .request(url: self.url, session: self.session)
+                .validate()
+                .response { response in
+            
+                    guard response.error == nil else {
+                        if let error = response.error as NSError? {
+                            resolver.set(error)
+                        }
+                        return
+                    }
+            
+                    do {
+                        guard let data = response.data else { throw CoreError.DataSerialization() }
+                        resolver.set(try self.decoder.decode(K.self, from: data))
```

Comment:
Why using directly a JSONDecoder instead of a generic `Mapper`? wouldn't this limit the usability of the class?